sunpy/data/data_manager/cache.py

```diff
@@ -40,56 +39,60 @@ def download(self, urls, namespace='', redownload=False):
         """
         Downloads the files from the urls.
 
-        The overall flow of this function is:
-            1. If ``redownload``: Download, update cache and return file path.
-            2. If not ``redownload``: Check cache,
-                i. If present in cache:
-                    - If cache has expired, remove the entry from cache, download and add to cache
-                    - If cache has not expired, return the path
-
         Parameters
         ----------
-        urls : `list` or `str`
+        urls : `list` of path-like or one path-like
             A list of urls or a single url.
-        redownload : `bool`
+            The list is for urls of the same file but from different sources.
+        namespace : `str`, optional
+            A namespace to be used for the file name.
+            Defaults to an empty string.
+        redownload : `bool`, optional
             Whether to skip cache and redownload.
+            Defaults to `False`.
 
         Returns
         -------
-        `pathlib.PosixPath`
+        `pathlib.Path`
             Path to the downloaded file.
         """
-        if isinstance(urls, str):
+        if isinstance(urls, str | Path):
             urls = [urls]
-        # Program flow
-        # 1. If redownload: Download, update cache and return file path
-        # 2. If not redownload: Check cache,
-        #    i. If present in cache:
-        #        - If cache expired, remove entry from cache, download and add to cache
-        #        - If cache not expired, return path
-        details = None
+        # Logic plan
+        # 1. Check if the file is present in cache by url
+        # 2. If present and it has not expired nor redownload, return the file path
+        # 3. If not present or present and (expired or redownload), download the file and update cache
+        #   a. If there is an error from the above steps, we will return the file from the cache if present
```

Comment:
Do we know for sure that the unit tests cover these steps?

Reply:
I think for now we cover 
1 )https://github.com/sunpy/sunpy/blob/main/sunpy/data/data_manager/tests/test_cache.py#L27-L31

3 a) https://github.com/sunpy/sunpy/blob/367d84e2c149879820af460bff7e801be6793dae/sunpy/data/data_manager/tests/test_cache.py#L50-L65